synapse/state/v2.py

```diff
@@ -284,10 +287,15 @@ async def _get_auth_chain_difference(
     #
     # Note: If the `event_map` is empty (which is the common case), we can do a
     # much simpler calculation.
-    difference_from_event_map = set()
     if event_map:
-        state_sets_ids = []
-        unpersisted_set_ids = []
+        # The list of state sets to pass to the store. This is the same as
+        # `state_sets` except with unpersisted events stripped out and replaced
+        # with persisted events in their auth chain.
```

Comment:
`state_sets` is a list of `(type, state_key) -> event_id` mappings, whereas here we just have the event ids, so this isn't quite right.

```suggestion
        # The list of state sets to pass to the store, where each state set is a set
        # of the event ids making up the state. This is similar to `state_sets`,
        # except that (a) we only have event ids, not the complete
        # ((type, state_key)->event_id) mappings; and (b) we have stripped out
        # unpersisted events and replaced them with the persisted events in 
        # their auth chain.
```

maybe?

Reply:
Yeah, that's better thanks :)